resources/developer-guides/daemon-rpc.md

```diff
@@ -1427,6 +1441,231 @@ $ curl http://127.0.0.1:18081/json_rpc -d '{"jsonrpc":"2.0","id":"0","method":"g
+* *merkle_tree_depth* usigned int; Auxilary proof-of-work merkle tree depth.
+* *aux_pow* - The array of aux_pow objects from the request.
+
+
+### **generateblocks**
+
+For use only in regression testing mode. Generate new blocks on your fake chain.
+
+Alias: *None*.
+
+Inputs:
+
+* *amount_of_blocks* - unsigned int; The count of blocks to generate.
+* *wallet_address* - string (base 58 encoded address); The wallet address used for coinbase rewards in the newly generated blocks.
+* *prev_block* - 32 byte hex string; The hash of the block to mine on top of.
+* *starting_nonce* - usigned int; The nonce you choose to begin with when "mining" these fake blocks.
+
+Outputs:
+
+* *height* - usigned int; The new chain height.
+* *blocks* - array of strings; Each string is the block id of each block generated.
+
+
+### **banned**
+
+Check if some host is banned by your node.
+
+Alias: *None*.
+
+Inputs:
+
+* *address* - string; The IP address to check.
+
+Outputs:
+
+* *banned* - boolean; True if banned, false if not.
+* *seconds* - unsigned int; Number of seconds this node is banned for.
+* *status* - string; General RPC error code. "OK" means everything looks good.
+
+
+### **prune_blockchain**
+
+Manage blockchain pruning.
+
+Alias: *None*.
+
+Inputs:
+
+* *check* - boolean; If true we only check if the node is using a pruned blockchain database. If false we will attempt to prune the blockchain database.
+
+Outputs:
+
+* *pruned* - boolean; Is this node using a pruned blockchain database.
+* *pruning_seed* - usigned int; This node's pruning seed number.
+
+
+### **flush_cache**
+
+Flush bad blocks and/or transactions from the node's cache.
+
+Alias: *None*.
+
+Inputs:
+
+* *bad_txs* - boolean; Flush bad transactions from node cache.
+* *bad_blocks* - boolean; Flush bad blocks from node cache.
+
+Outputs: *None*.
+
+---
+
+## RPC Pay Methods
+
+It is possible to configure a node to serve client wallets with blockchain data in exchange for proof-of-work that the node operator can use to try to collect miner rewards from.
+
+The following JSON RPC methods are specifically for facilitating this functionality.
+
+Every RPC Pay method request must include a client id (an elliptic curve public key).
+
+### **rpc_access_info**
+
+Fetch some general info about RPC Pay settings and status.
+
+Alias: *None*.
+
+Inputs: *None*.
+
+Outputs:
+
+* *hashing_blob* - string; A block template for the next block with the reward pointing toward the node operator, serialized to a hex string.
+* *seed_height* - usigned int; The height of the block to use as seed in Random-X proof-of-work.
+* *seed_hash* - string (32 byte hex); The id of the block to use as seed in Random-X proof-of-work.
+* *next_seed_hash* - string (32 byte hex); The id of the next block to use as seed in Random-X proof-of-work.
+* *cookie* - usigned int; The identifier cookie to keep track of the client.
+* *diff* unsigned int; The current network difficulty.
```

Comment:
Quite in general I ask myself whether it would be worth for this RPC documentation to not simply give *unsigned int*, but also the width, i.e. 32 for `cookie` but 64 for `diff`, but this probably goes beyond this PR.

Reply:
Yeah i would keep that outside this PR for now.